### PR TITLE
doc: doc update for oauth 2.0 support in Harvest

### PR DESCRIPTION
**Pull request by Hardikl**

### CODE REVIEW

File: docs/configure-harvest-basic.md [hunk 1]
```diff
@@ -329,7 +336,42 @@ password: mypassword
...
+#!/bin/bash
+# script requests an access token from the authorization server
+# authorization returns an access token to the script
+# script writes the YAML formatted authToken like so:
+cat << EOF
+authToken: $authToken
+EOF
+```
+
+`get_credentials` that outputs authToken in YAML format for interacting with Keycloak auth provider:
```
Comment (cgrinds) [inline docs/configure-harvest-basic.md:350]: Below are a couple of examples of how to create a credentials script that uses an OAuth2 workflow to authenticate with ONTAP or StorageGRID OAuth-enabled clusters.

??? note "These are examples that you will need to adapt to your environment."

    Example OAuth2 script authenticating with the Keycloak auth provider via `curl`. Uses [jq](https://github.com/jqlang/jq) to extract the token. This script outputs the authToken in YAML format.
    
    ```bash
    #!/bin/bash

    response=$(curl "http://{KEYCLOAK_IP:PORT}/realms/{REALM_NAME}/protocol/openid-connect/token" \
      --header "Content-Type: application/x-www-form-urlencoded" \
      --data-urlencode "grant_type=password" \
      --data-urlencode "username={USERNAME}" \
      --data-urlencode "password={PASSWORD}" \
      --data-urlencode "client_id={CLIENT_ID}" \
      --data-urlencode "client_secret={CLIENT_SECRET}")

    access_token=$(echo "$response" | jq -r '.access_token')

    cat << EOF
    authToken: $access_token
    EOF
    ```

    Example OAuth2 script authenticating with the Auth0 auth provider via `curl`. Uses [jq](https://github.com/jqlang/jq) to extract the token. This script outputs the authToken in YAML format.

    ```bash
    #!/bin/bash
    response=$(curl https://{AUTH0_TENANT_URL}/oauth/token \
      --header 'content-type: application/json' \
      --data '{"client_id":"{CLIENT_ID}","client_secret":"{CLIENT_SECRET}","audience":"{ONTAP_CLUSTER_IP}","grant_type":"client_credentials"')
    
    access_token=$(echo "$response" | jq -r '.access_token')

    cat << EOF
    authToken: $access_token
    EOF
    ```

Comment (Hardikl) [reply docs/configure-harvest-basic.md]: Done

File: docs/configure-harvest-basic.md [hunk 2]
```diff
@@ -347,29 +347,44 @@ authToken: $authToken
...
-curl "http://{KEYCLOAK_IP:PORT}/realms/{REAlM_NAME}/protocol/openid-connect/token" \
-  -X POST \
-  --header "Content-Type: application/x-www-form-urlencoded" \
-  --data-urlencode "grant_type=password" \
-  --data-urlencode "username={USERNAME}" \
-  --data-urlencode "password={PASSWORD}" \
-  --data-urlencode "client_id={CLIENT_ID}" \
-  --data-urlencode "client_secret={CLIENT_SECRET}" | jq | grep access_token | tr -d '"' | tr -d ',' | sed 's/access_token/authToken/'
-```
+Below are a couple of examples of how to create a credentials script that uses an OAuth2 workflow to authenticate with ONTAP or StorageGRID OAuth-enabled clusters.
```
Comment (cgrinds) [inline docs/configure-harvest-basic.md:350]: Below are a couple of OAuth2 credential script examples for authenticating with ONTAP or StorageGRID OAuth2-enabled clusters.

Comment (Hardikl) [reply docs/configure-harvest-basic.md:350]: Done

**Review by cgrinds — APPROVED** [review]

**Pull request merged**